tests/Fixture/Hooks/AddressAfterInstantiateHook.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the zenstruck/foundry package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Foundry\Tests\Fixture\Hooks;
+
+use Zenstruck\Foundry\Hooks\AfterInstantiate;
+use Zenstruck\Foundry\Hooks\AsAfterInstantiateFoundryHook;
+use Zenstruck\Foundry\Tests\Fixture\Entity\Address;
+
+#[AsAfterInstantiateFoundryHook(class: Address::class)]
+final class AddressAfterInstantiateHook
+{
+    /**
+     * @param AfterInstantiate<Address> $hook
+     */
+    public function __invoke(AfterInstantiate $hook): void
+    {
+        if ($hook->factory instanceof AddressFactoryWithHook) {
+            $hook->object->setCity(
+                'AddressAfterInitialisationHook '.$hook->object->getCity()
+            );
+        }
+    }
+}
```

Comment:
here is an example of a hook.

another syntax is possible:
```php
#[AsFoundryHook(event: AfterInstantiate::class, class: Address::class)]
final class AddressAfterInstantiateHook
{
    public function __invoke(AfterInstantiate $hook): void;
}
```
it would be clear then, that you should pass a `AfterInstantiate` to the `__invoke()` method